README.md

```diff
@@ -1,2 +1,20 @@
 # taar
 Telemetry-Aware Addon Recommender
+
+## How does it work?
+The recommendation strategy is implemented through the [RecommendationManager](taar/recommenders/recommendation_manager.py). Once a recommendation is requested for a specific [client id](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/data/common-ping.html), the recommender iterates through all the registered models (e.g. [CollaborativeRecommender](taar/recommenders/collaborative_recommender.py)), in the order they were registered in, and returns the results from the first model module that can perform a recommendation.
+
+Each module specify its own sets of rules and requirements and thus can decide if it can perform a recommendation independently from the other modules.
+
+### Supported models
+This is the ordered list of the currently supported models:
+
+| Order | Model | Description | Conditions | Generator job |
+|-------|-------|-------------|------------|---------------|
+| 1 | [Legacy](taar/recommenders/legacy_recommender.py) | recommends WebExtensions based on the reported and disabled legacy add-ons | Telemetry data is available for the user and the user has disabled add-ons| N/A |
+| 2 | [Collaborative](taar/recommenders/collaborative_recommender.py) | recommends add-ons based on add-ons installed by other users|Telemetry data is available for the user and the user has at least one enabled add-on|[source](https://github.com/mozilla/telemetry-batch-view/blob/master/src/main/scala/com/mozilla/telemetry/ml/AddonRecommender.scala)|
```

Comment:
Mention that this is an implementation of a collaborative filtering model (maybe also a wikipedia link)?